internal/overlord/servstate/manager_test.go

```diff
@@ -203,6 +212,45 @@ func (s *S) TestStartBadCommand(c *C) {
 	c.Assert(svc.Current, Equals, servstate.StatusInactive)
 }
 
+func (s *S) TestUserGroupFails(c *C) {
+	// Test with user and group will fail due to permission issues
+	var gotUid uint32
+	var gotGid uint32
+	restore := servstate.FakeSetCmdCredential(func(cmd *exec.Cmd, credential *syscall.Credential) {
+		gotUid = credential.Uid
+		gotGid = credential.Gid
+		cmd.SysProcAttr.Credential = credential
+	})
+	defer restore()
+
+	s.st.Lock()
+	ts, err := servstate.Start(s.st, []string{"test5"})
+	c.Check(err, IsNil)
+	chg := s.st.NewChange("test", "Start test")
+	chg.AddAll(ts)
+	s.st.Unlock()
+
+	s.ensure(c, 1)
+
+	s.st.Lock()
+	c.Check(chg.Status(), Equals, state.ErrorStatus)
+	c.Check(chg.Err(), ErrorMatches, `.*\n.*cannot start service: .* operation not permitted.*`)
```

Comment:
Does this test fail if run as root? We should probably at least skip it then?

Reply:
Good call. I added code to skip it if root.